angr/procedures/glibc/__libc_start_main.py

```diff
@@ -133,12 +133,6 @@ def run(self, main, argc, argv, init, fini):
         self.call(self.init, (self.argc, self.argv, self.envp), 'after_init')
 
     def after_init(self, main, argc, argv, init, fini, exit_addr=0):
-        if isinstance(self.state.arch, ArchAMD64):
```

Comment:
Can we make this logic a test case somewhere in angr? Otherwise we will break it soon (maybe after the next refactor).